File: snippets/js/node/src/live.ts
import {
    Client,
    ClientNotification,
    Filter,
    Kind,
    KindStandard,
    RelayUrl,
    ReqTarget,
    Timestamp,
} from "@nostrdevkit/nostr-sdk-node";

async function main() {
    const client = new Client();
    await client.addRelay(RelayUrl.parse("wss://relay.damus.io"));
    await client.connect();

    const notifications = client.notifications();
    const filter = new Filter()
        .kind(Kind.fromStd(KindStandard.TextNote))
        .since(Timestamp.now());
    const subscription = await client.subscribe(ReqTarget.auto([filter]));

    while (true) {
        const notification = await notifications.next();
        if (!notification) break;

        if (ClientNotification.NewEvent.instanceOf(notification)) {
            if (notification.inner.subscriptionId === subscription.id) {
                console.log(
                    `Received ${notification.inner.event.id().toBech32()} ` +
                    `from ${notification.inner.relayUrl}`,
                );
                break;
            }
        }
    }

    await client.unsubscribe(subscription.id);
}

await main();

File: snippets/kotlin/src/main/kotlin/Live.kt
package rust.nostr.snippets

import kotlinx.coroutines.runBlocking
import org.nostrdevkit.sdk.*

suspend fun receiveLiveEvent() {
    val client = Client()
    client.addRelay(RelayUrl.parse("wss://relay.damus.io"))
    client.connect()

    val notifications = client.notifications()
    val filter = Filter()
        .kind(Kind.fromStd(KindStandard.TEXT_NOTE))
        .since(Timestamp.now())
    val subscription = client.subscribe(ReqTarget.auto(listOf(filter)))

    while (true) {
        when (val notification = notifications.next() ?: break) {
            is ClientNotification.NewEvent -> {
                if (notification.subscriptionId == subscription.id) {
                    println("Received ${notification.event.id().toBech32()} from ${notification.relayUrl}")
                    break
                }
            }
            else -> Unit
        }
    }

    client.unsubscribe(subscription.id)
}

fun main() = runBlocking {
    receiveLiveEvent()
}
